app/services/species_uploader.rb

```diff
@@ -0,0 +1,13 @@
+class SpeciesUploader
+
+  def initialize json_file
+    @json_file = json_file
+  end
+
+  def run
+    JsonSpecies.to_a(@json_file).each do |json_specie|
+      json_specie_attrs = json_specie.attributes
+      Species.with_name(json_specie_attrs.name).first_or_create(json_specie_attrs).update(json_specie_attrs)
```

Comment:
Just another tinhy improvement, mode this line into a private method:

```ruby
JsonSpecies.to_a(@json_file).each do |json_specie|
   create_or_update_species(json_specie.attributes)
end

private

def create_or_update species json_specie_attributes
  Species.with_name(json_specie_attrs.name).first_or_create(json_specie_attrs).update(json_specie_attrs)
end
```

We save us a line :)